bls/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+pub use crate::pod::{
+    ProofOfPossession, ProofOfPossessionCompressed, Pubkey, PubkeyCompressed, Signature,
+    SignatureCompressed, BLS_PROOF_OF_POSSESSION_AFFINE_SIZE,
+    BLS_PROOF_OF_POSSESSION_COMPRESSED_SIZE, BLS_PUBLIC_KEY_AFFINE_SIZE,
+    BLS_PUBLIC_KEY_COMPRESSED_SIZE, BLS_SIGNATURE_AFFINE_SIZE, BLS_SIGNATURE_COMPRESSED_SIZE,
+};
+#[cfg(not(target_os = "solana"))]
+pub use crate::{
+    error::BlsError,
+    keypair::{PubkeyProjective, SecretKey, BLS_SECRET_KEY_SIZE},
+    proof_of_possession::ProofOfPossessionProjective,
+    signature::SignatureProjective,
+};
+#[cfg(not(target_os = "solana"))]
+use {
+    blstrs::{pairing, G1Affine, G2Projective},
+    group::prime::PrimeCurveAffine,
+};
+
+// TODO: add conversion between compressed and uncompressed representation of
+// signatures, pubkeys, and proof of possessions
+
+#[cfg(not(target_os = "solana"))]
+pub mod error;
+#[cfg(not(target_os = "solana"))]
+pub mod keypair;
+pub mod pod;
+#[cfg(not(target_os = "solana"))]
+pub mod proof_of_possession;
+#[cfg(not(target_os = "solana"))]
+pub mod signature;
+
+/// Domain separation tag used for hashing messages to curve points to prevent
+/// potential conflicts between different BLS implementations. This is defined
+/// as the ciphersuite ID string as recommended in the standard
+/// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-05#section-4.2.1.
+pub const HASH_TO_POINT_DST: &[u8] = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_";
+
+/// Domain separation tag used when hashing public keys to G2 in the proof of
+/// possession signing and verification functions. See
+/// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-05#section-4.2.3.
+pub const POP_DST: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_POP_";
+
+#[cfg(not(target_os = "solana"))]
+pub struct Bls;
+#[cfg(not(target_os = "solana"))]
+impl Bls {
```

Comment:
nit: I typically find it's clearer to avoid a lot of code in `lib.rs`. Since most of these are related to messages, maybe it can get moved to `message.rs`? Or maybe `pairing.rs`? But I don't love either of these, so we can also leave this as is if you prefer

Reply:
Ok that works